main.go

```diff
@@ -64,68 +50,77 @@ func applyVirtualMachineExport(vmNamespace, vmName string) error {
+	_, err := client.VirtualMachineExport(vmNamespace).Create(context.Background(), vmExport, metav1.CreateOptions{})
 	return err
 }
 
-func downloadVirtualMachineDiskImage(vmName, volumeName string) error {
-	log.Printf("Downloading disk image from %s Virtual Machine...\n", vmName)
-
-	cmd := exec.Command("usr/bin/virtctl", "vmexport", "download", vmName, "--vm", vmName, "--volume", volumeName, "--output", diskPath)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+func getRawDiskUrlFromVirtualMachineExport(client kubecli.KubevirtClient, vmNamespace, vmName, volumeName string) (string, error) {
+	log.Println("Waiting for VirtualMachineExport to be ready...")
 
-	if err := cmd.Run(); err != nil {
-		return err
+	vmExport, err := getVirtualMachineExportOnceReady(client, vmNamespace, vmName)
+	if err != nil {
+		return "", err
 	}
 
-	if fileInfo, err := os.Stat(diskPath); err != nil || fileInfo.Size() == 0 {
-		return fmt.Errorf("File does not exist or is empty.")
+	if vmExport.Status.Links == nil {
+		return "", fmt.Errorf("No links found in VirtualMachineExport status.")
 	}
 
-	log.Println("Download completed successfully.")
-	return nil
-}
+	if vmExport.Status.Links.Internal == nil {
+		return "", fmt.Errorf("No internal links found in VirtualMachineExport status.")
+	}
 
-func decompressVirtualMachineDiskImage() error {
-	log.Println("Decompressing downloaded disk image...")
+	for _, volume := range vmExport.Status.Links.Internal.Volumes {
+		if volumeName != volume.Name {
+			continue
+		}
 
-	diskImg, err := os.Open(diskPath)
-	if err != nil {
-		return err
+		for _, format := range volume.Formats {
+			if format.Format == v1beta1.KubeVirtRaw {
+				return format.Url, nil
+			}
+		}
 	}
-	defer diskImg.Close()
 
-	gzipReader, err := gzip.NewReader(diskImg)
-	if err != nil {
-		return err
-	}
-	defer gzipReader.Close()
+	return "", fmt.Errorf("Could not get raw disk URL from the VirtualMachineExport object.")
+}
 
-	newDiskImg, err := os.Create(diskPathDecompressed)
-	if err != nil {
-		return err
+func getVirtualMachineExportOnceReady(client kubecli.KubevirtClient, vmNamespace, vmName string) (*v1beta1.VirtualMachineExport, error) {
+	poller := func(context.Context) (bool, error) {
+		vmExport, err := client.VirtualMachineExport(vmNamespace).Get(context.Background(), vmName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		if vmExport.Status.Phase == v1beta1.Ready {
+			return true, nil
+		}
+		return false, nil
 	}
-	defer newDiskImg.Close()
 
-	_, err = io.Copy(newDiskImg, gzipReader)
+	err := wait.PollUntilContextTimeout(context.Background(), pollInterval, pollTimeout, true, poller)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Failed to wait for VirtualMachineExport to be ready: %v", err)
 	}
 
-	err = os.Chmod(diskPathDecompressed, 0666) // Grants read and write permission to everyone.
+	vmExport, err := client.VirtualMachineExport(vmNamespace).Get(context.Background(), vmName, metav1.GetOptions{})
```

Comment:
this call is not needed, since you can define the vmExport before the poller function and then the poller will populate the object.

Reply:
Yep thanks :)